FEATURE: Translate CR Nodes automatically.

mficzel opened the pull request:
Translate node properties while they are adopted to another language.

When nodes are copied (adopted) into another languge the fields can be translated automatically.

The following setting enables the translation of all inlineEditable properties: 

```yaml
CodeQ:
  DeepLTranslationHelper:
    nodeTranslations:
      translateInlineEditables: true
```

Other properties of type string can be translated aswell with the following configuration.

```yaml
Neos.Neos:Document:
   properties:
     title:
       options:
         deeplTranslate: true
```

In additions this PR:
- moves the caching to the eel helper 
- implements the deepl service with psr factories and request to avoid an explicit dependency to guzzle
- adds a Setting `CodeQ.DeepLTranslationHelper.DeepLService.useFreeApi: false` to switch between free and payed api
- adds a Setting `CodeQ.DeepLTranslationHelper.DeepLService.defaultOptions` to configure additional arguments for the translation

mficzel committed:
FEATURE: Translate CR Nodes on create.

This has still lots of limitations an probably needs tweaking.

rolandschuetz commented:
Wow, that was fast :)

Definitely needs some docs.
Let me know if I can help.

mficzel commented:
Am bit skeptical regarding performance because every translated property will yield a separate request.

Maybe the documents api could help here as this would support to translate multiple fields at once.

Or we trick a little and put all the properties of a node into a single html that gets translated and later split again. However would render the caching quite obsolete for this specific use case.

mficzel committed:
TASK: Move translation cache to the EelHelper

This is in preparation of translating multiple texts with one api call in the deepl service.

mficzel committed:
TASK: Adjust translation api to support the translation of multiple fragments with a single request.

mficzel commented:
@rolandschuetz some things that i still would like to improve are

- Some explicit mapping of dimension values to source and target language. Currently the first part of the language dimension value
- Some explicit configurations for glossaries (or probably additional request argument options based on language combination)
- Do nothing when no `language` dimension exists or make the whole dimension configurable.

Do you have specific ideas for that? 

PS: If you feel this totally is out of scope for your package just tell me and i will use the code otherwise. Could make sense to have a separate TranslationHelper and NodeTranslator package.

mficzel committed:
TASK: Use PSR7  and flow httpFactories

This avoids the direct dependency to guzzle but still uses psr factories and requests as much as possible.
Only the http timeout is set as curl option.

mficzel commented:
Close in favor of custom package

rolandschuetz commented:
@mficzel Sorry for the late response, I thought we might go over it together next week.
Was the delay for merging to long for you, or is your package structure very different?

mficzel commented:
The new package has very few similarities to yours and even my original pr:
https://github.com/sitegeist/Sitegeist.LostInTranslation 
Since this package also offers totally different features than yours this allows us to move faster independently.

rolandschuetz commented:
Looks cool.

So I will keep my package as EelHelper.. what I need, and keep your project in mind if I need that feature in another project :)
